src/kudu/consensus/raft_consensus.cc

```diff
@@ -3037,6 +3032,20 @@ Status RaftConsensus::SetCommittedConfigUnlocked(const RaftConfigPB& config_to_c
   return Status::OK();
 }
 
+void RaftConsensus::ScheduleTermAdvancementCallback(int64_t new_term) {
+  WARN_NOT_OK(
+      raft_pool_token_->SubmitFunc(
+        std::bind(&RaftConsensus::DoTermAdvancmentCallback,
+                  shared_from_this(),
+                  new_term)),
+      LogPrefixThreadSafe() + "Unable to run term advancement callback");
+}
+
+void RaftConsensus::DoTermAdvancmentCallback(int64_t new_term) {
+  // Simply execute the registered callback for term advancement.
+  tacb_(new_term);
```

Comment:
```suggestion
  if (tacb_)
    tacb_(new_term);
```

Reply:
Good point. However, both callbacks are private with public setters. These setters crash on the emptiness check like you had suggested. If you were considering the case where we have forgot to set, the default constructor for std::function initializes with empty function and the behaviour would be what you want anyway. Let me know if I missed something.

Reply:
Wondering if this will fail existing unit/integration tests in kudu? (Ref: https://en.cppreference.com/w/cpp/utility/functional/bad_function_call)

Reply:
It might fail new UTs in kudu. Diff updated.